Mathlib/FieldTheory/LinearDisjoint.lean

```diff
@@ -0,0 +1,361 @@
+
+/-- If `A` and `B` are linearly disjoint, then for any `F`-linearly independent families
+`{ u_i }`, `{ v_j }` of `A`, `B`, the products `{ u_i * v_j }`
+are linearly independent over `F`. -/
+theorem linearIndependent_mul (H : A.LinearDisjoint B)
+    {κ ι : Type*} {a : κ → A} {b : ι → B} (ha : LinearIndependent F a)
+    (hb : LinearIndependent F b) : LinearIndependent F fun (i : κ × ι) ↦ (a i.1).1 * (b i.2).1 :=
+  (linearDisjoint_iff'.1 H).linearIndependent_mul_of_flat_left ha hb
+
+/-- If `A` and `L` are linearly disjoint, then for any `F`-linearly independent families
+`{ u_i }`, `{ v_j }` of `A`, `L`, the products `{ u_i * v_j }`
+are linearly independent over `F`. -/
+theorem linearIndependent_mul' (H : A.LinearDisjoint L)
+    {κ ι : Type*} {a : κ → A} {b : ι → L} (ha : LinearIndependent F a)
+    (hb : LinearIndependent F b) :
+    LinearIndependent F fun (i : κ × ι) ↦ (a i.1).1 * algebraMap L E (b i.2) := by
+  have := Subalgebra.LinearDisjoint.linearIndependent_mul_of_flat_left H ha <| hb.map' _
+    (AlgEquiv.ofInjectiveField (IsScalarTower.toAlgHom F L E)).toLinearEquiv.ker
+  exact this
+
+/-- If there are `F`-bases `{ u_i }`, `{ v_j }` of `A`, `B`, such that the products
+`{ u_i * v_j }` are linearly independent over `F`, then `A` and `B` are linearly disjoint. -/
+theorem of_basis_mul {κ ι : Type*} (a : Basis κ F A) (b : Basis ι F B)
+    (H : LinearIndependent F fun (i : κ × ι) ↦ (a i.1).1 * (b i.2).1) : A.LinearDisjoint B :=
+  linearDisjoint_iff'.2 (.of_basis_mul _ _ a b H)
+
+/-- If there are `F`-bases `{ u_i }`, `{ v_j }` of `A`, `L`, such that the products
+`{ u_i * v_j }` are linearly independent over `F`, then `A` and `L` are linearly disjoint. -/
+theorem of_basis_mul' {κ ι : Type*} (a : Basis κ F A) (b : Basis ι F L)
+    (H : LinearIndependent F fun (i : κ × ι) ↦ (a i.1).1 * algebraMap L E (b i.2)) :
+    A.LinearDisjoint L := by
+  rw [linearDisjoint_iff]
+  exact .of_basis_mul _ _ a
+    (b.map (AlgEquiv.ofInjectiveField (IsScalarTower.toAlgHom F L E)).toLinearEquiv) H
+
+theorem of_le_left {A' : IntermediateField F E} (H : A.LinearDisjoint L)
+    (h : A' ≤ A) : A'.LinearDisjoint L :=
+  Subalgebra.LinearDisjoint.of_le_left_of_flat H h
+
+theorem of_le_right {B' : IntermediateField F E} (H : A.LinearDisjoint B)
+    (h : B' ≤ B) : A.LinearDisjoint B' :=
+  linearDisjoint_iff'.2 ((linearDisjoint_iff'.1 H).of_le_right_of_flat h)
+
+/-- Similar to `IntermediateField.LinearDisjoint.of_le_right` but this is for abstract fields. -/
+theorem of_le_right' (H : A.LinearDisjoint L) (L' : Type*) [Field L']
+    [Algebra F L'] [Algebra L' L] [IsScalarTower F L' L]
+    [Algebra L' E] [IsScalarTower F L' E] [IsScalarTower L' L E] : A.LinearDisjoint L' := by
+  refine Subalgebra.LinearDisjoint.of_le_right_of_flat H ?_
+  convert AlgHom.range_comp_le_range (IsScalarTower.toAlgHom F L' L) (IsScalarTower.toAlgHom F L E)
+  ext x
+  exact IsScalarTower.algebraMap_apply L' L E x
```

Comment:
No, it's not working, since I need to `AlgHom` are equal, but the result you mentioned only says two `RingHom` are equal. And it seems that we don't have result saying `Function.Injective AlgHom.toRingHom`.